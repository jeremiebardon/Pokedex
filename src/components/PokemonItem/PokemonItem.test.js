import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { test, expect } from 'jest-without-globals';

import PokemonsItem from './PokemonItem';

const props = {
  name: 'ivysaur',
  image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png',
  types: [
    {
      slot: 1,
      type: {
        name: 'grass',
        url: 'https://pokeapi.co/api/v2/type/12/',
      },
    },
    {
      slot: 2,
      type: {
        name: 'poison',
        url: 'https://pokeapi.co/api/v2/type/4/',
      },
    },
  ],
};

test('Render Item Element', () => {
  const component = render(
    <PokemonsItem name={props.name} image={props.image} types={props.types} />,
    { wrapper: MemoryRouter },
  );
  expect(component).toMatchSnapshot();
});

test('Render Text', () => {
  const component = render(
    <PokemonsItem name={props.name} image={props.image} types={props.types} />,
    { wrapper: MemoryRouter },
  );
  expect(component.getByText(props.name)).toBeInTheDocument();
});

test('Render Images', () => {
  const component = render(
    <PokemonsItem name={props.name} image={props.image} types={props.types} />,
    { wrapper: MemoryRouter },
  );
  expect(component.getByAltText(props.name)).toBeInTheDocument();
});
