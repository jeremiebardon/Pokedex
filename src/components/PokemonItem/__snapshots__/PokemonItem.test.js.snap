// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Item Element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/pokemons/ivysaur"
      >
        <div
          class="ant-card ant-card-bordered ant-card-hoverable"
          style="margin-bottom: 16px;"
        >
          <div
            class="ant-card-cover"
          >
            <img
              alt="ivysaur"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
            />
          </div>
          <div
            class="ant-card-body"
            style="padding: 8px; text-align: center;"
          >
            <h3
              style="font-weight: 800; text-transform: Capitalize; margin-bottom: 8px; font-size: 16px;"
            >
              ivysaur
            </h3>
            <span
              class="ant-tag"
            >
              grass
            </span>
            <span
              class="ant-tag"
            >
              poison
            </span>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/pokemons/ivysaur"
    >
      <div
        class="ant-card ant-card-bordered ant-card-hoverable"
        style="margin-bottom: 16px;"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="ivysaur"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
          />
        </div>
        <div
          class="ant-card-body"
          style="padding: 8px; text-align: center;"
        >
          <h3
            style="font-weight: 800; text-transform: Capitalize; margin-bottom: 8px; font-size: 16px;"
          >
            ivysaur
          </h3>
          <span
            class="ant-tag"
          >
            grass
          </span>
          <span
            class="ant-tag"
          >
            poison
          </span>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
